adapters/repos/db/vector/hnsw/visited/pool.go

```diff
@@ -17,54 +17,61 @@ import (
 
 type Pool struct {
 	sync.Mutex
-	listSize int
-	lists    []*List
+	listSetSize int
+	listSets    []ListSet
```

Comment:
same question here about the benefit of working on objects not on pointers to objects

Reply:
Using value objects is always faster than using pointer.
Each pointer means one more level of indirection to reach the actual object in memory.

Reply:
got it